lib/src/simple_op_heads_store.rs

```diff
@@ -83,24 +84,41 @@ impl OpHeadsStore for SimpleOpHeadsStore {
         Self::name()
     }
 
-    fn update_op_heads(&self, old_ids: &[OperationId], new_id: &OperationId) {
+    fn update_op_heads(
+        &self,
+        old_ids: &[OperationId],
+        new_id: &OperationId,
+    ) -> Result<(), OpHeadsStoreError> {
         assert!(!old_ids.contains(new_id));
         self.add_op_head(new_id);
```

Comment:
nit: `add_op_heads()` unwraps io::Error.

`remove_op_head()` can be more strict to propagate error except for `ENOENT`, but that can be addressed later.

Reply:
> remove_op_head() can be more strict to propagate error except for ENOENT

Oh, never mind. I misunderstood it intended to ignore double deletes.

Reply:
Good points. I added the error propagation from `add_op_heads()` and a TODO about making `remove_op_head()` more strict, since that's a bit more risky (but I agree it's correct).

Reply:
I added another commit to this PR for that.